frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx

```diff
@@ -1,21 +1,461 @@
+    getFieldWarning: jest.fn(),
+    getFieldsError: jest.fn(),
+    getFieldsValue: jest.fn(),
+    getInternalHooks: jest.fn(),
+    isFieldTouched: jest.fn(),
+    isFieldValidating: jest.fn(),
+    isFieldsTouched: jest.fn(),
+    isFieldsValidating: jest.fn(),
+    resetFields: jest.fn(),
+    scrollToField: jest.fn(),
+    setFields: jest.fn(),
+    setFieldsValue: jest.fn(),
+    submit: jest.fn(),
+    validateFields: jest.fn()
+  }]
+}));
 
 describe('TreeFormField component', () => {
-  const requiredProps = {};
+  const mockedTreeData = [
+    {
+      title: 'node1',
+      key: 'node1',
+      children: [],
+      parent: null,
+      properties: {
+        title: 'node1',
+        description: 'the node 1'
+      },
+      expanded: true,
+      isLeaf: false,
+    },
+    {
+      title: 'leaf1',
+      key: 'leaf1',
+      children: [],
+      parent: null,
+      properties: {
+        title: 'leaf1',
+        description: 'leaf 1 '
+      },
+      expanded: true,
+      isLeaf: true,
+    },
+  ];
 
-  const getComponent = (props?:any) => shallow((
+  const requiredProps = {
+    treeData: mockedTreeData,
+  };
+  
+  const getComponent = (props?:TreeProps) => render(
     <TreeManager
       {...requiredProps}
       {...props}
     />
-  ));
+  );
+  
+  // clears all mock artifacts to avoid transition from values and making all tests clean from the beginning
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
+  });
+
+  // NOTE: This component loads the tree data as soon as the tree data is ready.
+  // This fetch will trigger a state event (in this case setTreeData()).
+  // Because of this, we need to simulate this behaviour in an asynchronous fashion, hence the waitFor is used for all
+  // the following tests.
+  // After an event that triggers a state chnage, this is also being made.
+  // Without this, the tests will pass but a couple of warnings about state changes should be
+  // encapsulated in an act() method will appear
+  
+  it('defines the component', async () => {
+    const component =  getComponent(); 
+    await waitFor(() => expect(component).toBeDefined()); 
+  });
+
+  it('renders the component elements', async() => {
+    const { container } = getComponent();
+    await waitFor(() =>{
+      expect(container).toBeVisible();
+    });
+  });
+
+  it('renders the tree-manager-tree class', async() => {
+    const { container } = getComponent();
+    await waitFor(() =>{
+      expect(container.querySelectorAll('.tree-manager-tree').length).toBe(1);
+    });
+  });
+
+  it('does not render tree nodes when tree data does not contains information', async() => {
+    const { container } = getComponent({ treeData: [] });
+    await waitFor(() =>{
+      const treeFormFieldTree = container.querySelectorAll('.tree-manager-tree');
+      const treeFormFieldNode = treeFormFieldTree.item(0)?.querySelectorAll('.tree-manager-node');  
+      expect(treeFormFieldNode?.length).toBe(0);
+    });
+  });
+
```

Comment:
suggestion: 'renders tree nodes when data contains information' (the tree is rendered in any case, as the first test showed)